hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SubscriptionAssignmentRegistry.java

```diff
@@ -101,6 +94,38 @@ public SubscriptionAssignmentView createSnapshot() {
         return new SubscriptionAssignmentView(snapshot);
     }
 
+    private List<SubscriptionAssignment> readExistingAssignments() {
+        return subscriptionsCache.listActiveSubscriptionNames().stream().flatMap(subscriptionName -> {
+            String path = pathSerializer.serialize(subscriptionName);
+            List<String> children = new ArrayList<>();
+            try {
+                children.addAll(curator.getChildren().forPath(path));
+            } catch (Exception e) {
+                // ignore - should be fixed by the cache
+            }
+            return children.stream().map(node -> new SubscriptionAssignment(node, subscriptionName));
+        }).collect(Collectors.toList());
+    }
+
+    private void onAssignmentAdded(SubscriptionAssignment assignment) {
```

Comment:
That's true. I didn't consider checking because in consumer initiation those signals were compacted, however this is true that we should avoid issuing the signal if we already have the registry initialized with this assignment.